packages/block-library/src/search/view.js

```diff
@@ -0,0 +1,65 @@
+window.addEventListener( 'DOMContentLoaded', () => {
+	const hiddenClass = 'wp-block-search__searchfield-hidden';
+
+	Array.from(
+		document.getElementsByClassName(
+			'wp-block-search__button-behavior-expand'
+		)
+	).forEach( ( block ) => {
+		const searchField = block.querySelector( '.wp-block-search__input' );
+		const searchButton = block.querySelector( '.wp-block-search__button' );
+		const activeElement = block.ownerDocument.activeElement;
+
+		const ariaLabel = searchButton.getAttribute( 'aria-label' );
+		const id = searchField.getAttribute( 'id' );
+
+		const toggleSearchField = ( showSearchField ) => {
+			if ( showSearchField ) {
+				searchField.removeAttribute( 'aria-hidden' );
+				searchField.removeAttribute( 'tabindex' );
+				searchButton.removeAttribute( 'aria-expanded' );
+				searchButton.removeAttribute( 'aria-controls' );
+				searchButton.removeAttribute( 'aria-label' );
+
+				return block.classList.remove( hiddenClass );
+			}
+
+			searchField.setAttribute( 'aria-hidden', 'true' );
+			searchField.setAttribute( 'tabindex', '-1' );
+			searchButton.setAttribute( 'aria-expanded', 'false' );
+			searchButton.setAttribute( 'aria-controls', id );
+			searchButton.setAttribute( 'aria-label', ariaLabel );
```

Comment:
This causes problems if `searchButton` never had an `aria-label`. It will set the `aria-label` attribute to the string value "null", overriding any visible label the button might have. We should either check that `ariaLabel` has a value before doing this, or give `ariaLabel` a fallback value of "" above on line 13.

Reply:
If this script is running, `searchButton` will always have an `aria-label`, which is defined [here](https://github.com/WordPress/gutenberg/pull/50487/files#diff-ef7ce82e804956a7d5c001fefbceabf1820872331c6282c9d359cac1e7390734R128-R130).